rest_framework_swagger/tests.py

```diff
@@ -1608,7 +1663,7 @@ def post(self, request, *args, **kwargs):
 
         class_introspector = self.make_introspector(SerializedAPI)
         introspector = APIViewMethodIntrospector(class_introspector, 'POST')
-        generator = DocumentationGenerator()
+        generator = DocumentationGenerator(for_user=self.user)
```

Comment:
Perhaps factor `DocumentationGenerator(for_user=self.user)` out to `self.get_documentation_generator()`